sw/device/lib/sdk/spi/spi_sdk.c

```diff
@@ -756,12 +756,12 @@ spi_codes_e spi_set_slave(spi_t* spi)
     // Build the HAL configopts to be set based on our slave
     spi_configopts_t config = {
         .clkdiv   = clk_div,
-        .cpha     = bitfield_read(spi->slave.data_mode, BIT_MASK_1, DATA_MODE_CPHA_OFFS),
-        .cpol     = bitfield_read(spi->slave.data_mode, BIT_MASK_1, DATA_MODE_CPOL_OFFS),
         .csnidle  = spi->slave.csn_idle,
-        .csnlead  = spi->slave.csn_lead,
         .csntrail = spi->slave.csn_trail,
-        .fullcyc  = spi->slave.full_cycle
+        .csnlead  = spi->slave.csn_lead,
```

Comment:
why has the order changed?

Reply:
In C++ you need the follow the same order that was set on the structs definition, otherwise you get at error